docs/design-docs/benchmarks.md

```diff
@@ -0,0 +1,262 @@
+---
+id: benchmarks
+---
+
+# Benchmarks Design
+
+## Goals
+
+The design of `benchmarks` has the following goals to achieve:
+
+**High Code Quality**
+* Extract the common code into the base class, reduce the efforts to maintain different benchmarks.
+
+**Good Extensibility**
+* Avoid modifying existing code when adding new benchmarks by using registration mechanism.
+* Support pre-definition of benchmarks' settings, and support benchmark registration with different settings.
```

Comment:
Support pre-definition of benchmarks' settings, and benchmark registration with different settings.